File: AuctionWebsite/AuctionService/src/main/java/com/auctionservice/service/AuctionService.java
package com.auctionservice.service;

import java.util.List;

import com.auctionservice.dto.AuctionDTO;
import com.auctionservice.entity.Auction;

public interface AuctionService {
	String createAuction(Auction auction);

	Auction updateAuction(int id, Auction auction);

	List<Auction> getAllAuction();

	Auction getAuctionById(int id);

	String deleteAuction(int id);

	String endAuction(int id);

	String startAuction(int id);

	AuctionDTO getByAuctionId(int id);

	void updateHighestBid(int id, double price);
}

File: auction/src/services/auctionService.js
// src/services/auctionService.js

import apiClient from '../api/apiClient';

const auctionService = {
  getAllAuctions: () => apiClient.get('/auction/all'),
  getAuctionById: (id) => apiClient.get(`/auction/${id}`),
  createAuction: (auctionData) => apiClient.post('/auction/create', auctionData),
  updateAuction: (id, auctionData) => apiClient.put('/auction/update', auctionData, { params: { id } }),
  deleteAuction: (id) => apiClient.delete('/auction/delete', { params: { id } }),
  startAuction: (id) => apiClient.post('/auction/start', null, { params: { id } }),
  endAuction: (id) => apiClient.post('/auction/end', null, { params: { id } }),
  updateHighestBid: (id, price) => apiClient.post(`/auction/${id}/${price}`)
};

export default auctionService;
